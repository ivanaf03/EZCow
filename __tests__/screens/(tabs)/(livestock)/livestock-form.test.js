import React from "react";

import { render, fireEvent, waitFor } from "@testing-library/react-native";
import { router } from "expo-router";
import { Alert } from "react-native";

import LivestockForm from "../../../../app/(tabs)/(livestock)/livestock-form";
import {
  insertCow,
  getAvailableCowCodeByUserId,
} from "../../../../model/cow";

jest.mock("expo-router", () => ({
  router: {
    replace: jest.fn(),
  },
}));

jest.mock("../../../../store/user-provider", () => ({
  useUser: jest.fn(() => ({
    user: { id: "1", name: "TestUser", email: "test@test.com" },
  })),
}));

jest.mock("../../../../model/cow", () => ({
  insertCow: jest.fn(),
  getAvailableCowCodeByUserId: jest.fn(() => ["1234", "5678"]),
  getCowIdByCode: jest.fn(() => "ES1234"),
}));

jest.mock("@fortawesome/react-native-fontawesome", () => ({
  FontAwesomeIcon: () => null,
}));

jest.spyOn(Alert, "alert");

describe("LivestockForm", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("should render correctly", async () => {
    const tree = render(<LivestockForm />);
    await waitFor(() => expect(getAvailableCowCodeByUserId).toHaveBeenCalled());
    expect(tree).toMatchSnapshot();
  });

  it("should show an error if fields are empty", async () => {
    const { getByTestId } = render(<LivestockForm />);
    await waitFor(() => expect(getAvailableCowCodeByUserId).toHaveBeenCalled());
    const handleAddCowButton = getByTestId("handle-add-cow-button");
    fireEvent.press(handleAddCowButton);
    expect(Alert.alert).toHaveBeenCalledWith(
      "Error",
      "Por favor, rellena todos los campos."
    );
  });

  it("should show an error if code is empty", async () => {
    const { getByText, getByPlaceholderText, getByTestId } = render(
      <LivestockForm />
    );
    await waitFor(() => expect(getAvailableCowCodeByUserId).toHaveBeenCalled());

    fireEvent.changeText(getByPlaceholderText("Nombre"), "Vaca Test");
    fireEvent.changeText(getByPlaceholderText("Raza"), "Holstein");
    fireEvent(getByText("Género:"), "onValueChange", "Femenino");
    fireEvent(getByText("Madre:"), "onValueChange", "1234");
    fireEvent(getByText("Fase:"), "onValueChange", "Ternero");

    const handleAddCowButton = getByTestId("handle-add-cow-button");
    fireEvent.press(handleAddCowButton);

    expect(Alert.alert).toHaveBeenCalledWith(
      "Error",
      "Por favor, rellena todos los campos."
    );
  });

  it("should show an error if name is empty", async () => {
    const { getByText, getByPlaceholderText, getByTestId } = render(
      <LivestockForm />
    );
    await waitFor(() => expect(getAvailableCowCodeByUserId).toHaveBeenCalled());

    fireEvent.changeText(getByPlaceholderText("Código"), "COW001");
    fireEvent.changeText(getByPlaceholderText("Raza"), "Holstein");
    fireEvent(getByText("Género:"), "onValueChange", "Femenino");
    fireEvent(getByText("Madre:"), "onValueChange", "1234");
    fireEvent(getByText("Fase:"), "onValueChange", "Ternero");

    const handleAddCowButton = getByTestId("handle-add-cow-button");
    fireEvent.press(handleAddCowButton);

    expect(Alert.alert).toHaveBeenCalledWith(
      "Error",
      "Por favor, rellena todos los campos."
    );
  });

  it("should show an error if breed is empty", async () => {
    const { getByText, getByPlaceholderText, getByTestId } = render(
      <LivestockForm />
    );
    await waitFor(() => expect(getAvailableCowCodeByUserId).toHaveBeenCalled());

    fireEvent.changeText(getByPlaceholderText("Código"), "COW001");
    fireEvent.changeText(getByPlaceholderText("Nombre"), "Vaca Test");
    fireEvent(getByText("Género:"), "onValueChange", "Femenino");
    fireEvent(getByText("Madre:"), "onValueChange", "1234");
    fireEvent(getByText("Fase:"), "onValueChange", "Ternero");

    const handleAddCowButton = getByTestId("handle-add-cow-button");
    fireEvent.press(handleAddCowButton);

    expect(Alert.alert).toHaveBeenCalledWith(
      "Error",
      "Por favor, rellena todos los campos."
    );
  });

  it("should call insertCow on valid submission", async () => {
    const { getByText, getByPlaceholderText, getByTestId } = render(
      <LivestockForm />
    );
    await waitFor(() => expect(getAvailableCowCodeByUserId).toHaveBeenCalled());

    fireEvent.changeText(getByPlaceholderText("Código"), "COW001");
    fireEvent.changeText(getByPlaceholderText("Nombre"), "Vaca Test");
    fireEvent.changeText(getByPlaceholderText("Raza"), "Holstein");
    fireEvent(getByText("Género:"), "onValueChange", "Femenino");
    fireEvent(getByText("Madre:"), "onValueChange", "1234");
    fireEvent(getByText("Fase:"), "onValueChange", "Ternero");

    fireEvent.press(getByTestId("handle-add-cow-button"));

    await waitFor(() => {
      expect(insertCow).toHaveBeenCalledWith(
        "COW001",
        "Vaca Test",
        expect.stringMatching(/\d{4}-\d{2}-\d{2}/),
        "Femenino",
        "Holstein",
        "Ternero",
        "1",
        "ES1234"
      );
      expect(router.replace).toHaveBeenCalledWith("livestock");
    });
  });

  it("should navigate to livestock when pressing livestock button", async () => {
    const { getByTestId } = render(<LivestockForm />);
    await waitFor(() => expect(getAvailableCowCodeByUserId).toHaveBeenCalled());
    const handleAddCowButton = getByTestId("livestock-button");
    fireEvent.press(handleAddCowButton);
    expect(router.replace).toHaveBeenCalledWith("livestock");
  });
});
