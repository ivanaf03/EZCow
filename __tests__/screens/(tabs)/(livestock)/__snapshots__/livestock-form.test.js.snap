// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LivestockForm should render correctly 1`] = `
<RCTScrollView
  style={
    [
      {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
      },
      {
        "backgroundColor": "#1c1c1c",
      },
    ]
  }
>
  <View>
    <View
      style={
        [
          {
            "marginTop": 32,
          },
          {
            "paddingBottom": 16,
            "paddingTop": 16,
          },
          {
            "marginLeft": 16,
            "marginRight": 16,
          },
          {
            "alignItems": "center",
          },
          {
            "borderLeftWidth": 2,
          },
          {
            "borderRightWidth": 2,
          },
          {
            "borderBottomColor": "#6bc0eb",
            "borderLeftColor": "#6bc0eb",
            "borderRightColor": "#6bc0eb",
            "borderTopColor": "#6bc0eb",
          },
          {
            "borderBottomLeftRadius": 16,
            "borderBottomRightRadius": 16,
            "borderTopLeftRadius": 16,
            "borderTopRightRadius": 16,
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "width": "90%",
            },
          ]
        }
      >
        <View
          style={
            [
              {
                "marginBottom": 8,
                "marginLeft": 8,
                "marginRight": 8,
                "marginTop": 8,
              },
              {
                "marginTop": -4,
              },
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#ffffff",
                },
                {
                  "fontSize": 14,
                  "lineHeight": 20,
                },
                {
                  "fontFamily": "Nunito-Medium",
                },
                [
                  {
                    "marginTop": 4,
                  },
                ],
              ]
            }
          >
            Código de chapa
            :
          </Text>
          <TextInput
            onChangeText={[Function]}
            placeholder="Código"
            style={
              [
                {
                  "paddingBottom": 8,
                  "paddingLeft": 8,
                  "paddingRight": 8,
                  "paddingTop": 8,
                },
                {
                  "backgroundColor": "#ffffff",
                },
                {
                  "borderBottomWidth": 2,
                  "borderLeftWidth": 2,
                  "borderRightWidth": 2,
                  "borderTopWidth": 2,
                },
                {
                  "borderBottomColor": "#2254ba",
                  "borderLeftColor": "#2254ba",
                  "borderRightColor": "#2254ba",
                  "borderTopColor": "#2254ba",
                },
                {
                  "borderBottomLeftRadius": 16,
                  "borderBottomRightRadius": 16,
                  "borderTopLeftRadius": 16,
                  "borderTopRightRadius": 16,
                },
                {
                  "fontFamily": "Nunito-Medium",
                },
                [
                  {
                    "marginTop": 4,
                  },
                ],
              ]
            }
          />
        </View>
      </View>
      <View
        style={
          [
            {
              "width": "90%",
            },
            [
              {
                "marginTop": 16,
              },
            ],
          ]
        }
      >
        <View
          style={
            [
              {
                "marginBottom": 8,
                "marginLeft": 8,
                "marginRight": 8,
                "marginTop": 8,
              },
              {
                "marginTop": -4,
              },
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#ffffff",
                },
                {
                  "fontSize": 14,
                  "lineHeight": 20,
                },
                {
                  "fontFamily": "Nunito-Medium",
                },
                [
                  {
                    "marginTop": 4,
                  },
                ],
              ]
            }
          >
            Nombre de vaca
            :
          </Text>
          <TextInput
            onChangeText={[Function]}
            placeholder="Nombre"
            style={
              [
                {
                  "paddingBottom": 8,
                  "paddingLeft": 8,
                  "paddingRight": 8,
                  "paddingTop": 8,
                },
                {
                  "backgroundColor": "#ffffff",
                },
                {
                  "borderBottomWidth": 2,
                  "borderLeftWidth": 2,
                  "borderRightWidth": 2,
                  "borderTopWidth": 2,
                },
                {
                  "borderBottomColor": "#2254ba",
                  "borderLeftColor": "#2254ba",
                  "borderRightColor": "#2254ba",
                  "borderTopColor": "#2254ba",
                },
                {
                  "borderBottomLeftRadius": 16,
                  "borderBottomRightRadius": 16,
                  "borderTopLeftRadius": 16,
                  "borderTopRightRadius": 16,
                },
                {
                  "fontFamily": "Nunito-Medium",
                },
                [
                  {
                    "marginTop": 4,
                  },
                ],
              ]
            }
          />
        </View>
      </View>
      <View
        style={
          [
            {
              "width": "90%",
            },
            [
              {
                "marginTop": 16,
              },
            ],
          ]
        }
      >
        <View
          style={
            [
              {
                "marginBottom": 8,
                "marginLeft": 8,
                "marginRight": 8,
                "marginTop": 8,
              },
              {
                "marginTop": -4,
              },
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#ffffff",
                },
                {
                  "fontSize": 14,
                  "lineHeight": 20,
                },
                {
                  "fontFamily": "Nunito-Light",
                },
                [
                  {
                    "marginTop": 4,
                  },
                ],
              ]
            }
          >
            Fecha de incorporación
            :
          </Text>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "paddingBottom": 8,
                  "paddingLeft": 8,
                  "paddingRight": 8,
                  "paddingTop": 8,
                },
                {
                  "backgroundColor": "#ffffff",
                },
                {
                  "borderBottomWidth": 2,
                  "borderLeftWidth": 2,
                  "borderRightWidth": 2,
                  "borderTopWidth": 2,
                },
                {
                  "borderBottomColor": "#2254ba",
                  "borderLeftColor": "#2254ba",
                  "borderRightColor": "#2254ba",
                  "borderTopColor": "#2254ba",
                },
                {
                  "borderBottomLeftRadius": 16,
                  "borderBottomRightRadius": 16,
                  "borderTopLeftRadius": 16,
                  "borderTopRightRadius": 16,
                },
                {
                  "fontFamily": "Nunito-Medium",
                },
                [
                  {
                    "marginTop": 4,
                  },
                ],
              ]
            }
            testID="date-picker-button"
          >
            <Text>
              30/3/2025
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          [
            {
              "width": "90%",
            },
            [
              {
                "marginTop": 16,
              },
            ],
          ]
        }
      >
        <View
          style={
            [
              {
                "marginBottom": 8,
                "marginLeft": 8,
                "marginRight": 8,
                "marginTop": 8,
              },
              {
                "marginTop": -4,
              },
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#ffffff",
                },
                {
                  "fontSize": 14,
                  "lineHeight": 20,
                },
                {
                  "fontFamily": "Nunito-Light",
                },
                [
                  {
                    "marginTop": 4,
                  },
                ],
              ]
            }
          >
            Género
            :
          </Text>
          <View
            style={
              [
                {
                  "backgroundColor": "#ffffff",
                },
                {
                  "borderBottomWidth": 2,
                  "borderLeftWidth": 2,
                  "borderRightWidth": 2,
                  "borderTopWidth": 2,
                },
                {
                  "borderBottomColor": "#2254ba",
                  "borderLeftColor": "#2254ba",
                  "borderRightColor": "#2254ba",
                  "borderTopColor": "#2254ba",
                },
                {
                  "borderBottomLeftRadius": 24,
                  "borderBottomRightRadius": 24,
                  "borderTopLeftRadius": 24,
                  "borderTopRightRadius": 24,
                },
                {
                  "fontFamily": "Nunito-Medium",
                },
                [
                  {
                    "marginTop": 4,
                  },
                ],
              ]
            }
          >
            <View>
              <RNCPicker
                items={
                  [
                    {
                      "label": "Masculino",
                      "testID": undefined,
                      "textColor": undefined,
                      "value": "Masculino",
                    },
                    {
                      "label": "Femenino",
                      "testID": undefined,
                      "textColor": undefined,
                      "value": "Femenino",
                    },
                  ]
                }
                numberOfLines={1}
                onChange={[Function]}
                selectedIndex={0}
                style={
                  [
                    {
                      "height": 216,
                    },
                    undefined,
                  ]
                }
              />
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          [
            {
              "width": "90%",
            },
            [
              {
                "marginTop": 16,
              },
            ],
          ]
        }
      >
        <View
          style={
            [
              {
                "marginBottom": 8,
                "marginLeft": 8,
                "marginRight": 8,
                "marginTop": 8,
              },
              {
                "marginTop": -4,
              },
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#ffffff",
                },
                {
                  "fontSize": 14,
                  "lineHeight": 20,
                },
                {
                  "fontFamily": "Nunito-Light",
                },
                [
                  {
                    "marginTop": 4,
                  },
                ],
              ]
            }
          >
            Fase
            :
          </Text>
          <View
            style={
              [
                {
                  "backgroundColor": "#ffffff",
                },
                {
                  "borderBottomWidth": 2,
                  "borderLeftWidth": 2,
                  "borderRightWidth": 2,
                  "borderTopWidth": 2,
                },
                {
                  "borderBottomColor": "#2254ba",
                  "borderLeftColor": "#2254ba",
                  "borderRightColor": "#2254ba",
                  "borderTopColor": "#2254ba",
                },
                {
                  "borderBottomLeftRadius": 24,
                  "borderBottomRightRadius": 24,
                  "borderTopLeftRadius": 24,
                  "borderTopRightRadius": 24,
                },
                {
                  "fontFamily": "Nunito-Medium",
                },
                [
                  {
                    "marginTop": 4,
                  },
                ],
              ]
            }
          >
            <View>
              <RNCPicker
                items={
                  [
                    {
                      "label": "Ternero",
                      "testID": undefined,
                      "textColor": undefined,
                      "value": "Ternero",
                    },
                    {
                      "label": "Adulto",
                      "testID": undefined,
                      "textColor": undefined,
                      "value": "Adulto",
                    },
                  ]
                }
                numberOfLines={1}
                onChange={[Function]}
                selectedIndex={0}
                style={
                  [
                    {
                      "height": 216,
                    },
                    undefined,
                  ]
                }
              />
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          [
            {
              "width": "90%",
            },
            [
              {
                "marginTop": 16,
              },
            ],
          ]
        }
      >
        <View
          style={
            [
              {
                "marginBottom": 8,
                "marginLeft": 8,
                "marginRight": 8,
                "marginTop": 8,
              },
              {
                "marginTop": -4,
              },
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#ffffff",
                },
                {
                  "fontSize": 14,
                  "lineHeight": 20,
                },
                {
                  "fontFamily": "Nunito-Medium",
                },
                [
                  {
                    "marginTop": 4,
                  },
                ],
              ]
            }
          >
            Raza
            :
          </Text>
          <TextInput
            onChangeText={[Function]}
            placeholder="Raza"
            style={
              [
                {
                  "paddingBottom": 8,
                  "paddingLeft": 8,
                  "paddingRight": 8,
                  "paddingTop": 8,
                },
                {
                  "backgroundColor": "#ffffff",
                },
                {
                  "borderBottomWidth": 2,
                  "borderLeftWidth": 2,
                  "borderRightWidth": 2,
                  "borderTopWidth": 2,
                },
                {
                  "borderBottomColor": "#2254ba",
                  "borderLeftColor": "#2254ba",
                  "borderRightColor": "#2254ba",
                  "borderTopColor": "#2254ba",
                },
                {
                  "borderBottomLeftRadius": 16,
                  "borderBottomRightRadius": 16,
                  "borderTopLeftRadius": 16,
                  "borderTopRightRadius": 16,
                },
                {
                  "fontFamily": "Nunito-Medium",
                },
                [
                  {
                    "marginTop": 4,
                  },
                ],
              ]
            }
          />
        </View>
      </View>
      <View
        style={
          [
            {
              "width": "90%",
            },
            [
              {
                "marginTop": 16,
              },
            ],
          ]
        }
      >
        <View
          style={
            [
              {
                "marginBottom": 8,
                "marginLeft": 8,
                "marginRight": 8,
                "marginTop": 8,
              },
              {
                "marginTop": -4,
              },
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#ffffff",
                },
                {
                  "fontSize": 14,
                  "lineHeight": 20,
                },
                {
                  "fontFamily": "Nunito-Light",
                },
                [
                  {
                    "marginTop": 4,
                  },
                ],
              ]
            }
          >
            Madre
            :
          </Text>
          <View
            style={
              [
                {
                  "backgroundColor": "#ffffff",
                },
                {
                  "borderBottomWidth": 2,
                  "borderLeftWidth": 2,
                  "borderRightWidth": 2,
                  "borderTopWidth": 2,
                },
                {
                  "borderBottomColor": "#2254ba",
                  "borderLeftColor": "#2254ba",
                  "borderRightColor": "#2254ba",
                  "borderTopColor": "#2254ba",
                },
                {
                  "borderBottomLeftRadius": 24,
                  "borderBottomRightRadius": 24,
                  "borderTopLeftRadius": 24,
                  "borderTopRightRadius": 24,
                },
                {
                  "fontFamily": "Nunito-Medium",
                },
                [
                  {
                    "marginTop": 4,
                  },
                ],
              ]
            }
          >
            <View>
              <RNCPicker
                items={
                  [
                    {
                      "label": "Desconocida",
                      "testID": undefined,
                      "textColor": undefined,
                      "value": "Desconocida",
                    },
                    {
                      "label": "1234",
                      "testID": undefined,
                      "textColor": undefined,
                      "value": "1234",
                    },
                    {
                      "label": "5678",
                      "testID": undefined,
                      "textColor": undefined,
                      "value": "5678",
                    },
                  ]
                }
                numberOfLines={1}
                onChange={[Function]}
                selectedIndex={0}
                style={
                  [
                    {
                      "height": 216,
                    },
                    undefined,
                  ]
                }
              />
            </View>
          </View>
        </View>
        <Text
          style={
            [
              {
                "color": "#ffffff",
              },
              {
                "paddingLeft": 16,
                "paddingRight": 16,
              },
              {
                "fontSize": 14,
                "lineHeight": 20,
              },
              {
                "fontFamily": "Nunito-LightItalic",
              },
            ]
          }
        >
          Marcar como desconocida si el animal proviene de otra explotación o si la madre no está registrada en la explotación.
        </Text>
      </View>
      <View
        style={
          [
            {
              "marginTop": 24,
            },
            {
              "width": "90%",
            },
            [
              {
                "marginTop": 16,
              },
            ],
          ]
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "backgroundColor": "#272727",
              },
              {
                "borderRightWidth": 8,
              },
              {
                "borderLeftWidth": 8,
              },
              {
                "borderBottomColor": "#2254ba",
                "borderLeftColor": "#2254ba",
                "borderRightColor": "#2254ba",
                "borderTopColor": "#2254ba",
              },
              {
                "borderBottomLeftRadius": 24,
                "borderBottomRightRadius": 24,
                "borderTopLeftRadius": 24,
                "borderTopRightRadius": 24,
              },
              {
                "paddingBottom": 16,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 16,
              },
              {
                "marginBottom": 8,
                "marginLeft": 8,
                "marginRight": 8,
                "marginTop": 8,
              },
            ]
          }
          testID="handle-add-cow-button"
        >
          <Text
            style={
              [
                {
                  "color": "#ffffff",
                },
                {
                  "fontSize": 18,
                  "lineHeight": 28,
                },
                {
                  "textAlign": "center",
                },
                {
                  "fontFamily": "Nunito-Medium",
                },
                {
                  "marginLeft": 8,
                  "marginRight": 8,
                },
              ]
            }
          >
            Añadir
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        [
          {
            "flexDirection": "row",
          },
          {
            "justifyContent": "center",
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "marginBottom": 16,
              "marginLeft": 16,
              "marginRight": 16,
              "marginTop": 16,
            },
            {
              "width": "80%",
            },
          ]
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "backgroundColor": "#272727",
              },
              {
                "borderRightWidth": 8,
              },
              {
                "borderLeftWidth": 8,
              },
              {
                "borderBottomColor": "#2254ba",
                "borderLeftColor": "#2254ba",
                "borderRightColor": "#2254ba",
                "borderTopColor": "#2254ba",
              },
              {
                "borderBottomLeftRadius": 24,
                "borderBottomRightRadius": 24,
                "borderTopLeftRadius": 24,
                "borderTopRightRadius": 24,
              },
              {
                "paddingBottom": 16,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 16,
              },
              {
                "marginBottom": 8,
                "marginLeft": 8,
                "marginRight": 8,
                "marginTop": 8,
              },
            ]
          }
          testID="livestock-button"
        >
          <Text
            style={
              [
                {
                  "color": "#ffffff",
                },
                {
                  "fontSize": 18,
                  "lineHeight": 28,
                },
                {
                  "textAlign": "center",
                },
                {
                  "fontFamily": "Nunito-Medium",
                },
                {
                  "marginLeft": 8,
                  "marginRight": 8,
                },
              ]
            }
          >
            <View
              style={
                [
                  {
                    "flexDirection": "column",
                  },
                  {
                    "alignItems": "center",
                  },
                ]
              }
            >
              <View
                style={
                  [
                    {
                      "paddingLeft": 8,
                      "paddingRight": 8,
                    },
                  ]
                }
              />
              <Text
                style={
                  [
                    {
                      "color": "#ffffff",
                    },
                    {
                      "fontSize": 18,
                      "lineHeight": 28,
                    },
                    {
                      "fontFamily": "Nunito-Medium",
                    },
                  ]
                }
              >
                Ver censo
              </Text>
            </View>
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
